File: discord_bot/webhook_reactions/release_event.py
import logging
import discord

from discord_bot.webhook_reactions.reactions import RPG_NOTIFICATIONS_CHANNEL_ID

log = logging.getLogger(__name__)

bot_instance: discord.Client | None = None


def set_bot_instance(bot: discord.Client):
    global bot_instance
    bot_instance = bot


async def post_new_release(url: str, release_url: str, release_name: str, release_body: str, tag_name: str):
    if bot_instance is None:
        log.error("Bot instance not set")
        return
    channel = bot_instance.get_channel(int(RPG_NOTIFICATIONS_CHANNEL_ID))
    if not isinstance(channel, discord.TextChannel):
        log.error("Could not find RPG notifications channel")
        return

    embed = discord.Embed(
        title="A new release has just been posted",
        description=f"Check out the latest release for Supergoon RPG with tag {tag_name}\nPlay the emscripten build here https://escapethefate.supergoon.com or the dev build here https://escapethefate-dev.supergoon.com",
        color=0x00FF00,
    )
    embed.add_field(name="Release URL (downloads and notes)", value=release_url, inline=True)
    embed.add_field(name="Name", value=release_name, inline=True)
    embed.add_field(name="Release Body", value=release_body, inline=False)
    embed.add_field(name="Repository Url", value=url, inline=False)
    embed.add_field(name="Post issues here.", value=f"{url}/issues", inline=False)

    await channel.send(embed=embed)

File: discord_bot/webhook_reactions/release_event.go
package webhook_reactions

import (
	"fmt"
	"log"

	"github.com/bwmarrin/discordgo"

	discordbot "sgg-discord-bot/discord_bot"
)

func PostNewRelease(url string, releaseURL string, releaseName string, releaseBody string, tagName string) {
	bot := discordbot.GetCurrentBot()
	if bot == nil {
		log.Println("Bot instance not set")
		return
	}

	embed := &discordgo.MessageEmbed{
		Title:       "A new release has just been posted",
		Description: fmt.Sprintf("Check out the latest release for Supergoon RPG with tag %s\nPlay the emscripten build here https://escapethefate-dev.supergoon.com", tagName),
		Color:       0x00FF00,
		Fields: []*discordgo.MessageEmbedField{
			{Name: "Release URL (downloads and notes)", Value: releaseURL, Inline: true},
			{Name: "Name", Value: releaseName, Inline: true},
			{Name: "Release Body", Value: releaseBody, Inline: false},
			{Name: "Repository Url", Value: url, Inline: false},
			{Name: "Post issues here.", Value: fmt.Sprintf("%s/issues", url), Inline: false},
		},
	}

	_, err := bot.ChannelMessageSendEmbed(RPGNotificationsChannelID, embed)
	if err != nil {
		log.Printf("Error sending release message: %v", err)
	}
}
